src/pyfreeleh/row/gsheet.py

```diff
@@ -1,412 +1,458 @@
-        self._query = QueryBuilder(c1_col_mapping)
+        self._query = GoogleSheetQueryBuilder()
 
     def where(self, condition: str, *args: Any) -> "DeleteStmt":
-        self._query.where(condition, *args)
+        """Filter the rows that we're going to delete.
+
+        The given `condition` will be used as the WHERE clause on the final query. You can use `"?"` placeholder
+        inside the condition and will be replaced with the actual value given in the `*args` variadic parameter
+        based on their appearance ordering.
+
+        Args:
+            condition: conditions of the data that we're going to delete.
+            *args: list of arguments that will be used to fill in the placeholders in the given `condition`.
+
+        Returns:
+            DeleteStmt: the delete statement with the given where condition applied.
+
+        Examples:
+            To apply "WHERE age > 10" filter on the delete statement:
+
+            >> store.delete().where("age > ?", 10).execute()
+            10
+        """
+        mapped_condition = map_conditions(self._mapper, condition)
+        self._query.where(mapped_condition, *args)
         return self
 
     def execute(self) -> int:
-        location = str(
-            A1Range(self._sheet_name, A1CellSelector.from_rc(1, 2), A1CellSelector.from_rc(column=len(self._columns)))
-        )
-        query = self._query.build_select([self.ROW_IDX_FIELD]).replace('"', '""')
-        formula = self.INDICES_FORMULA_TEMPLATE.format(location, location, query)
+        """Execute the delete statement.
 
-        result = self._wrapper.update_rows(self._spreadsheet_id, self._scratchpad_cell, [[formula]])
-        update_candidate_indices = [int(idx) for idx in result.updated_values[0][0].split(",") if idx]
+        Returns:
+            int: number of rows deleted.
+        """
+        # TODO(fata.nugraha): assumption pk is in the first cell
+        affected_rows = self._sheet_session.query(self._query.build_select(["A"]))
+        update_candidate_indices = [int(row["A"]) for row in affected_rows]
 
         requests = []
         for row_idx in update_candidate_indices:
             row_selector = A1CellSelector.from_rc(row=row_idx)
-            requests.append(A1Range(self._sheet_name, start=row_selector, end=row_selector))
+            requests.append(A1Range(self._sheet_session.sheet_name, start=row_selector, end=row_selector))
 
-        self._wrapper.clear(self._spreadsheet_id, requests)
+        self._sheet_session.clear(requests)
 
         return len(update_candidate_indices)
 
 
-class GoogleSheetRowStore:
-    SCRATCHPAD_BOOKED_VALUE = "BOOKED"
-    SCRATCHPAD_SUFFIX = "_scratch"
-    TS_COLUMN_NAME = "_ts"
-
+class GoogleSheetRowStore(Generic[T]):
     def __init__(
         self,
         auth_client: GoogleAuthClient,
         spreadsheet_id: str,
         sheet_name: str,
-        columns: List[str],
-        codec: Codec = BasicCodec(),
+        object_klass: Type[T],
```

Comment:
Is it better to use `cls` instead of `klass`? I think `klass` is more Java than Python.

Reply:
Done.